jetty-websocket/jetty-websocket-api/src/main/java/org/eclipse/jetty/websocket/api/Session.java

```diff
@@ -164,7 +164,11 @@ default WebSocketPolicy getPolicy()
 
     /**
      * Suspend the incoming read events on the connection.
-     *
+     * <p>
+     * This should be called during the processing of a frame or message to atomically
+     * suspend read events before the next frame is received. Calling suspend outside of
+     * this can result in a frame being received shortly after the suspend call.
+     * </p>
```

Comment:
Not good javadoc.
Start by saying what the method does: stops new messages from being delivered.
Then say how it is best used: if called from scope of message handler the call takes effect immediately
The say other behaviour. if called outside scope... then call may take effect after 1 more frame/message is delivered.

Reply:
It does start with what the method does "Suspend the incoming read events on the connection" but I can change this to something like "suspend incoming frames from being delivered".

Should I give some indication of the approximate time the frame could be delivered after, just saying may take effect after 1 more frame is delivered seems a bit vague. But we know the its a very small timing window so it would be delivered essentially immediately after suspend is called.

Reply:
saw your other comment about how the timing doesn't matter and that the API frame buffer should be removed. So I wont specify anything about the timing window of this in the javadoc.